File: OptiPlanBackend/OptiPlanBackend/Models/TeamMembership.cs
using OptiPlanBackend.Enums;

namespace OptiPlanBackend.Models
{
    public class TeamMembership
    {
        public Guid Id { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; }

        public Guid TeamId { get; set; }
        public Team Team { get; set; }

        public TeamRole Role { get; set; } = TeamRole.TeamMember;
        public MembershipStatus Status { get; set; } = MembershipStatus.Pending;
        public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
    }
}

File: optiplan-frontend/src/app/models/team-membership.ts
import { MembershipStatus } from "./enums/membership-status";
import { TeamRole } from "./enums/team-role";
import { Team } from "./team";
import { User } from "./user";


export interface TeamMembership {
  id: string;
  userId: string;
  user: User;
  teamId: string;
  team: Team;
  role: TeamRole;
  status: MembershipStatus;
  joinedAt: string;
}
